Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -1856,15 +1866,38 @@ - (void)collectionView:(JSQMessagesCollectionView *)collectionView
                             }
                         }
                     }
+                } else if ([messageItem.media isKindOfClass:[AttachmentPointerAdapter class]]) {
+                    AttachmentPointerAdapter *attachmentPointerAdadpter = (AttachmentPointerAdapter *)messageItem.media;
+
+                    TSAttachmentPointer *pointer = attachmentPointerAdadpter.attachmentPointer;
+                    if (pointer.state == TSAttachmentPointerStateFailed) {
```

Comment:
Does this "tap on download"?  Should it present an action sheet with "try again/cancel"? Just asking.